File: docs/examples/teaching/06-validation/recipe.py
from gramlot.page import WebPage


class Page(WebPage):
    def main(self, root):
        root.p("Enter 3 to 8 characters, then leave the field.")
        root.data("validation.code", "ABC")
        field = root.textBox(value="^validation.code", lbl="Code")
        field.validate(notnull=True, len="3:8")

File: docs/examples/teaching/06-validation/recipe.js
export function build(root) {
    root.p('Enter 3 to 8 characters, then leave the field.');
    root.data('validation.code', 'ABC');
    const field = root.textBox({value: '^validation.code', lbl: 'Code'});
    field.validate({notnull: true, len: '3:8'});
}
